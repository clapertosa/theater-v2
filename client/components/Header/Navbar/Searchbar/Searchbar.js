import React from "react";
import styled from "styled-components";

const Container = styled.div`
  border-radius: 10px;
  width: 100%;
  min-width: 15rem;
`;

const Input = styled.input`
  height: 2rem;
  width: inherit;
  text-indent: 30px;
  font-size: 1.2rem;
`;

const MagnifyingGlass = styled.i`
  position: absolute;
  font-size: 1.5rem;
`;

const Searchbar = () => {
  return (
    <Container>
      <MagnifyingGlass className="icon-search" />
      <Input type="text" placeholder="Search Movies or Series" />
    </Container>
  );
};

export default Searchbar;
