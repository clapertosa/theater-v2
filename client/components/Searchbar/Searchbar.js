import React, { useState } from "react";
import styled from "styled-components";
import Results from "./Results/Results";

const Container = styled.div`
  position: relative;
  border-radius: 10px;
  width: 100%;
`;

const Input = styled.input`
  height: 2rem;
  width: inherit;
  text-indent: 30px;
  font-size: 1.2rem;
`;

const MagnifyingGlass = styled.i`
  position: absolute;
  font-size: 1.5rem;
  color: black;
`;

const Searchbar = () => {
  const [query, setQuery] = useState("");
  const [showResults, setShowResults] = useState(false);
  const [results, setResults] = useState([]);

  const onChangeHandler = e => {
    setQuery(e.target.value);
  };

  return (
    <Container>
      <MagnifyingGlass className="icon-search" />
      <Input
        type="text"
        placeholder="Search"
        value={query}
        onChange={onChangeHandler}
      />
      <Results showResults={showResults} results={results} />
    </Container>
  );
};

export default Searchbar;
