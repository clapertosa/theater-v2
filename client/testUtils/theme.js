export default {
  colors: {
    gunMetal: "#2b2d42",
    gray: "#8d99ae",
    white: "#edf2f4",
    red: "#ef233c",
    candy: "#d90429"
  },
  navbarHeight: "4rem",
  mediaQueryMinWidth: "55rem"
};
