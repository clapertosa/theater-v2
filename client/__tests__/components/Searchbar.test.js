import { MockedProvider } from "react-apollo/test-utils";
import { ApolloConsumer } from "react-apollo";
import { ThemeProvider } from "styled-components";
import theme from "../../testUtils/theme";
import wait from "../../testUtils/wait";
import { shallow, mount } from "enzyme";
import Searchbar, { SEARCH_QUERY } from "../../components/Searchbar/Searchbar";

describe("SearchBar component", () => {
  let wrapper;
  let apolloClient;
  const mocks = [
    {
      request: {
        query: SEARCH_QUERY,
        variables: {
          query: "Meet Joe Black"
        }
      },
      result: {
        data: {
          search: {
            results: {
              id: 297,
              name: null,
              original_name: null,
              title: "Meet Joe Black",
              original_title: "Meet Joe Black",
              media_type: "movie",
              poster_path: "/s5maEXmEsVZYUx6dWc11GhL0zww.jpg",
              profile_path: null
            }
          }
        }
      }
    }
  ];

  beforeEach(() => {
    wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ThemeProvider theme={theme}>
          <ApolloConsumer>
            {client => {
              apolloClient = client;
              return <Searchbar />;
            }}
          </ApolloConsumer>
        </ThemeProvider>
      </MockedProvider>
    );
  });
  it("renders correctly", async () => {
    await wait();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it("queries with ApolloClient", async () => {
    const res = await apolloClient.query({
      query: SEARCH_QUERY,
      variables: { query: "Meet Joe Black" }
    });

    expect(res.data.search.results.title).toEqual("Meet Joe Black");
  });
});
