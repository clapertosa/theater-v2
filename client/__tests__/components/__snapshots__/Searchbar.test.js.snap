// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders correctly 1`] = `ReactWrapper {}`;
